File: tests/corpus/safe/parameterized-queries.ts
// Safe: Parameterized SQL queries
// Expected: NO detections

import { db } from './database';

export async function getUser(userId: number) {
  const result = await db.query('SELECT * FROM users WHERE id = $1', [userId]);  // Safe
  return result.rows[0];
}

export async function searchProducts(term: string) {
  const result = await db.query(
    'SELECT * FROM products WHERE name ILIKE $1',  // Safe
    [`%${term}%`]
  );
  return result.rows;
}

File: tests/corpus/safe/parameterized-queries.py
# Safe: Parameterized SQL queries
# Expected: NO detections

import sqlite3

def get_user(user_id: int):
    conn = sqlite3.connect('users.db')
    cursor = conn.cursor()
    cursor.execute("SELECT * FROM users WHERE id = ?", (user_id,))  # Safe: parameterized
    return cursor.fetchone()

def search_users(name: str):
    cursor = conn.cursor()
    cursor.execute("SELECT * FROM users WHERE name LIKE ?", (f"%{name}%",))  # Safe: parameterized
    return cursor.fetchall()

def insert_user(name: str, email: str):
    cursor.execute(
        "INSERT INTO users (name, email) VALUES (?, ?)",  # Safe: parameterized
        (name, email)
    )
